pre_commit/editor.py

```diff
@@ -0,0 +1,167 @@
+        allowed_option('--cleanup')
+        allowed_option('--pathspec-from-file')
+        allowed_flag('--pathspec-file-nul')
+
+        # == Disallowed arguments ==
+        # -m <msg>, --message=<msg>  # Alternate means of supplying a commit message.
+        # -F <file>, --file=<file>  # Alternate means of supplying a commit message.
+        # -C <commit>, --reuse-message=<commit>  # Alternate means of supplying a commit message.
+        # -c <commit>, --reedit-message=<commit>  # Alternate means of supplying a commit message.
+
+        # --fixup=<commit>  # We don't support automatically constructing a fixup commit message.
+        # --squash=<commit>  # We don't support automatically constructing a squash commit message.
+        # --dry-run  # No commit actually made.
+        # --short  # Implies --dry-run.
+        # --porcelain  # Implies --dry-run.
+        # --long  # Implies --dry-run.
+        # -z, --null  # Intended to be used with --short or --porcelain.
+        # -t <file>, --template=<file>  # We don't support template files.
+        # -s, --signoff  # We don't support adding a signoff line.
+        # -n, --no-verify  # We shouldn't be called if this is passed.
+        # -e, --edit  # We don't support editing a commit message supplied from other sources.
+        # --no-edit  # Explicitly requests not launching an editor.
+        # -u[<mode>], --untracked-files[=<mode>]  # We don't support showing untracked files differently.
+        # -v, --verbose  # We don't support verbose git status.
+        # --no-status  # We don't support not including the status.
+        # -S, --gpg-sign[=<keyid>], --no-gpg-sign  # Optional arg is annoying to support; rarely used.
+
+        parser.add_argument('pathspec', nargs='+')
+        parser.parse_args(git_invocation)
+
+        # Parse succeeded -- no unknown args.
+        return True
+    except ParseFailed:
+        return False
+
+def _is_editor_script_configured() -> bool:
+    editor_script_path = git.get_editor_script_path()
+    local_git_editor = _get_local_git_editor()
+    return (bool(local_git_editor)
+            and local_git_editor[0] == editor_script_path
+            and Path(editor_script_path).exists()
+           )
+
+def _edit_commit_message(template: str) -> None:
+    if not COMMIT_MESSAGE_DRAFT_PATH.exists():
+        COMMIT_MESSAGE_DRAFT_PATH.parent.mkdir(parents=True, exist_ok=True)
+        COMMIT_MESSAGE_DRAFT_PATH.write_text(template)
+    else:
+        # Update commit draft with new status
+        commit_draft = COMMIT_MESSAGE_DRAFT_PATH.read_text()
+        commit_draft = '\n'.join(line for line in commit_draft.splitlines() if not line.startswith('#'))
+        COMMIT_MESSAGE_DRAFT_PATH.write_text(commit_draft + template)
+    git_editor = _get_global_git_editor()  # Doesn't run in this repo, so the concurrency won't cause git lock errors.
+    with monitor.trace('precommit.editor'):
+        with open('/dev/tty') as stdin:
```

Comment:
![what-sorcery-is-this](https://user-images.githubusercontent.com/463696/129287637-83d5fe60-48e6-44c4-aa50-8f1a43d33dfb.jpg)

Reply:
I'll, uh, add a comment with a bit more info.